Client/Frontend/Menu/MenuPresentationAnimator.swift

```diff
@@ -0,0 +1,146 @@
+        self.presenting = true
+        return self
+    }
+
+    func animationControllerForDismissedController(dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        self.presenting = false
+        return self
+    }
+}
+
+extension MenuPresentationAnimator {
+    private func animateWithMenu(menu: MenuViewController, browserViewController bvc: BrowserViewController, transitionContext: UIViewControllerContextTransitioning) {
+        let leftViews: [UIView]?
+        let rightViews: [UIView]?
+        let sourceView: UIView?
+        if let toolbar = bvc.toolbar {
+            leftViews = [toolbar.backButton, toolbar.forwardButton]
+            rightViews = [toolbar.stopReloadButton, toolbar.shareButton]
+            sourceView = toolbar.menuButton
+        } else {
+            sourceView = nil
+            leftViews = nil
+            rightViews = nil
+        }
+
+        self.animateWithMenu(menu, baseController: bvc, viewsToAnimateLeft: leftViews, viewsToAnimateRight: rightViews, sourceView: sourceView, withTransitionContext: transitionContext)
+    }
+
+    private func animateWithMenu(menu: MenuViewController, tabTrayController ttc: TabTrayController, transitionContext: UIViewControllerContextTransitioning) {
+        
+    }
+
+    private func animateWithMenu(menuController: MenuViewController, baseController: UIViewController, viewsToAnimateLeft: [UIView]?, viewsToAnimateRight: [UIView]?, sourceView: UIView?, withTransitionContext transitionContext: UIViewControllerContextTransitioning) {
+
+        guard let container = transitionContext.containerView() else { return }
+
+        let menuView = menuController.view
+        let bottomView = baseController.view
+
+        // Insert tab tray below the browser and force a layout so the collection view can get it's frame right
+        if presenting {
+            container.insertSubview(menuView, belowSubview: bottomView)
+            menuView.layoutSubviews()
+        }
+
+        let menuViewSnapshot: UIView
+        if presenting {
+            menuViewSnapshot = menuView.snapshotViewAfterScreenUpdates(true)
+            menuViewSnapshot.frame = menuView.frame
+            menuViewSnapshot.transform = CGAffineTransformMakeScale(0.0, 1.0)
+            menuViewSnapshot.alpha = 0
+            menuView.backgroundColor = menuView.backgroundColor?.colorWithAlphaComponent(0.0)
+            menuView.addSubview(menuViewSnapshot)
+        } else {
+            menuViewSnapshot = menuView.snapshotViewAfterScreenUpdates(false)
+            menuViewSnapshot.frame = menuView.frame
+            container.insertSubview(menuViewSnapshot, aboveSubview: menuView)
+            menuView.hidden = true
+        }
+
+        let offstageLeft = CGAffineTransformMakeTranslation(-150, 0)
```

Comment:
Nit: Can we change this to be more flexible? Maybe something like 1/2 of screen or toolbar width?